beta-src/src/state/game/game-api-slice.ts

```diff
@@ -380,22 +424,26 @@ const gameApiSlice = createSlice({
     processUnitClick(state, clickData) {
       const {
         data: {
-          data: { contextVars },
-        },
-        order: {
-          inProgress,
-          method,
-          onTerritory,
-          orderID,
-          toTerritory,
-          type,
-          unitID,
+          data: { contextVars, currentOrders },
         },
+        order: { inProgress, method, onTerritory, orderID, type, unitID },
         ownUnits,
+        ordersMeta,
       } = current(state);
       if (contextVars?.context?.orderStatus) {
         const orderStates = getOrderStates(contextVars?.context?.orderStatus);
-        if (orderStates.Ready) {
+        const clickedUnit = clickData.payload.unitID;
+        const unitOrderFiltered = currentOrders?.filter(
+          (o) => o.unitID === clickedUnit,
+        );
+
+        if (
+          orderStates.Ready ||
+          (unitOrderFiltered?.length &&
+            ordersMeta[unitOrderFiltered[0]?.id].update?.type === "Disband" &&
+            !ordersMeta[unitOrderFiltered[0]?.id].allowedBorderCrossings
+              ?.length)
```

Comment:
shouldn't we need to loop through ordersMeta instead of checking only the 1st item?

Reply:
unitOrderFiltered would be an array of one current order if that current order belongs to the unit clicked. Order is not always in progress and in this case, we dont want it to begin because this unit is disbanded automatically and has nowhere to go. Im using the currentOrder belonging to the unit clicked to check its ordersMeta and see if it has anywhere to go and is disbanded